File: Music.Client.Android/Java/short/app/src/main/java/com/zitherharpmusic/zhmshort/ui/notification/NotificationFragment.java
package com.zitherharpmusic.zhmshort.ui.notification;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.zitherharpmusic.zhmshort.MainActivity;
import com.zitherharpmusic.zhmshort.R;
import com.zitherharpmusic.zhmshort.model.User;

public class NotificationFragment extends Fragment {
    private User user;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        user = ((MainActivity) requireActivity()).getUser();
        if (user.isLoggedIn()) {
            if (user.getArtists().size() > 0) {
                return inflater.inflate(R.layout.fragment_notification, container, false);
            } else {
                return inflater.inflate(R.layout.fragment_empty, container, false);
            }
        } else {
            return inflater.inflate(R.layout.fragment_login, container, false);
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        if (user.isLoggedIn()) {
            if (user.getArtists().size() > 0) {
                RecyclerView recyclerView = view.findViewById(R.id.recycler_view);
                recyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));
                recyclerView.setAdapter(new NotificationAdapter(this, user));
            } else {
                TextView textView = view.findViewById(R.id.empty);
                textView.setText(getString(R.string.no_notification));
            }
        } else {
            // TODO: login fragment
        }
    }
}

File: Music.Client.Android/Kotlin/Music/short/src/main/java/com/zitherharp/music/shorts/ui/main/NotificationFragment.kt
package com.zitherharp.music.shorts.ui.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.zitherharp.music.shorts.databinding.NotificationFragmentBinding
import com.zitherharp.music.ui.adapter.FragmentStateAdapter
import com.zitherharp.music.ui.fragment.EmptyFragment

class NotificationFragment: Fragment() {
    private lateinit var binding: NotificationFragmentBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
        NotificationFragmentBinding.inflate(inflater, container, false).apply {
            binding = this
        }.root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        with(binding) {
            NotificationAdapter(this@NotificationFragment,
                arrayOf("Cập nhật"))
                .attach(notificationTabLayout, notificationViewPager)
        }
    }

    inner class NotificationAdapter(fragment: Fragment, tabNames: Array<String>): FragmentStateAdapter(fragment, tabNames) {
        override fun createFragment(position: Int): Fragment {
            when (position) {
            }
            return EmptyFragment()
        }
    }
}